src/higley_lab_to_nwb/lohani_2022/interfaces/lohani_2022_visual_stimulus_interface.py

```diff
@@ -0,0 +1,121 @@
+
+
+def _test_sonpy_installation() -> None:
+    get_package(
+        package_name="sonpy",
+        excluded_python_versions=["3.10", "3.11"],
+        excluded_platforms_and_python_versions=dict(darwin=dict(arm=["3.8", "3.9", "3.10", "3.11"])),
+    )
+
+
+def get_streams(file_path: FilePathType) -> List[str]:
+    """Return a list of channel names as set in the recording extractor."""
+    r = io.CedIO(filename=file_path)
+    signal_channels = r.header["signal_channels"]
+    stream_ids = signal_channels["id"]
+    stream_names = signal_channels["name"]
+    return stream_ids, stream_names
+
+
+class Lohani2022VisualStimulusInterface(BaseDataInterface):
+    """
+    Data interface class for converting Spike2 visual stimulus signals from CED (Cambridge Electronic
+    Design) using the :py:class:`~spikeinterface.extractors.CedRecordingExtractor`."""
+
+    display_name = "Spike2 Recording"
+    associated_suffixes = (".smr", ".smrx")
+    info = "Interface for Spike2 analogue signals from CED (Cambridge Electronic Design)."
+
+    def __init__(
+        self,
+        spike2_file_path: FilePathType,
+        csv_file_path: FilePathType,
+        stream_id: str,
+        verbose: bool = True,
+    ):
+        """
+         Parameters
+        ----------
+        spike2_file_path : FilePathType
+            Path to .smr or .smrx file.
+        csv_file_path : FilePathType
+            Path to .csv file for visual stimulus characterization.
+        verbose : bool, default: True
+        """
+        _test_sonpy_installation()
+
+        super().__init__(
+            spike2_file_path=spike2_file_path,
+            csv_file_path=csv_file_path,
+            stream_id=stream_id,
+            verbose=verbose,
+        )
+
+    def get_event_times_from_ttl(self, rising: bool = True):
+        extractor = CedRecordingExtractor(
+            file_path=str(self.source_data["spike2_file_path"]), stream_id=self.source_data["stream_id"]
+        )
+        times = extractor.get_times()
+        traces = extractor.get_traces()
+        if rising:
+            event_times = get_rising_frames_from_ttl(traces)
+        else:
+            event_times = get_falling_frames_from_ttl(traces)
+
+        return times[event_times]
+
+    def get_stimulus_feature(self, column_index):
+        feature = pd.read_csv(self.source_data["csv_file_path"], usecols=column_index)
+        return feature.to_numpy()
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False) -> None:
+
+        intervals_table = TimeIntervals(
+            name="VisualStimulus",
+            description=f"Intervals for each visual stimulus presentation",
+        )
+
+        intervals_table.add_column(name="contrast", description="Contrast of the visual stimulus image.")
+        contrasts = self.get_stimulus_feature(column_index=[0])
+        intervals_table.add_column(name="orientation", description="Contrast of the visual stimulus image.")
```

Comment:
For orientation I immediately think of left/right etc. can you help me understand why it refers to contrast?

Reply:
Ah it's a copy paste

Reply:
Yes, good catch!